File: src/main/java/tk/skulk/plugin/core/extensions/foodbehaviour/listeners/CookieResetInsomniaListener.java
package tk.skulk.plugin.core.extensions.foodbehaviour.listeners;

import org.bukkit.Material;
import org.bukkit.Statistic;
import org.bukkit.event.EventHandler;
import org.bukkit.event.EventPriority;
import org.bukkit.event.player.PlayerItemConsumeEvent;
import org.jetbrains.annotations.NotNull;
import tk.skulk.plugin.core.extensions.foodbehaviour.FoodBehaviourExtension;
import tk.skulk.plugin.wrapper.BaseListener;

import static tk.skulk.plugin.helpers.MiniMessageHelper.sendMessage;

public final class CookieResetInsomniaListener extends BaseListener<FoodBehaviourExtension> {
    public CookieResetInsomniaListener(final @NotNull FoodBehaviourExtension extension) {
        super(extension);
    }

    @EventHandler(priority = EventPriority.LOWEST, ignoreCancelled = true)
    public void onPlayerItemConsume(final @NotNull PlayerItemConsumeEvent event) {
        if (event.getItem().getType() != Material.COOKIE) {
            return;
        }

        final var player = event.getPlayer();
        player.setStatistic(Statistic.TIME_SINCE_REST, 0);

        sendMessage(player, "gold", '✓', "Your insomnia has been reset.");
    }
}

File: src/main/kotlin/tk/skulk/plugin/core/extensions/foodBehaviour/listeners/CookieResetInsomniaListener.kt
package tk.skulk.plugin.core.extensions.foodBehaviour.listeners

import org.bukkit.Material
import org.bukkit.Statistic
import org.bukkit.event.EventHandler
import org.bukkit.event.EventPriority
import org.bukkit.event.player.PlayerItemConsumeEvent
import tk.skulk.plugin.core.SNPlugin
import tk.skulk.plugin.core.extensions.foodBehaviour.FoodBehaviourExtension
import tk.skulk.plugin.framework.Listener
import tk.skulk.plugin.util.sendMessage

class CookieResetInsomniaListener(extension: FoodBehaviourExtension) : Listener<SNPlugin, FoodBehaviourExtension>(
    extension
) {
    @EventHandler(priority = EventPriority.MONITOR, ignoreCancelled = true)
    fun onPlayerItemConsume(event: PlayerItemConsumeEvent) {
        if (event.item.type != Material.COOKIE) {
            return
        }

        val player = event.player

        if (player.getStatistic(Statistic.TIME_SINCE_REST) < 100) {
            return
        }

        event.player.setStatistic(Statistic.TIME_SINCE_REST, 0)

        event.player.sendMessage("green", '✓', "Your insomnia has been reset.")
    }
}
